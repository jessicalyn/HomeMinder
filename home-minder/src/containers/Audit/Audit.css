.audit-container {
  background-color: white;
  border: solid 2px black;
  border-radius: 2rem;
  width: 75%;
  margin: auto;
  margin-top: 2rem;
  padding: 1.5rem
}

.audit-cards-container {
  display: flex;
  flex-wrap: wrap;
}

h2 {
  margin-top: 0
}