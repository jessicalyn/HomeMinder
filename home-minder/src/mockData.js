export const mockHomeData = [
  { id: "JK1234",
    room: "Kitchen",
    category: "Filters",
    name: "Refrigerator Water Filter",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "6 months",
    userScheduled: "",
    type: "Replace"
  },
  { id: "IF1234",
    room: "Kitchen",
    category: "Sponges",
    name: "Sponge",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "2-4 weeks",
    userScheduled: "",
    type: "Replace"
  },
  { id: "PF123",
    room: "Bathroom",
    category: "Toothbrush",
    name: "Toothbrush",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "1-3 months",
    userScheduled: "",
    type: "Replace"
  },
]