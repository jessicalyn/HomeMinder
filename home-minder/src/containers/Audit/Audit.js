import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Audit.css';
import AuditCard from '../../components/AuditCard/AuditCard';
import PropTypes from 'prop-types';

export class Audit extends Component {

  render() {
    const { items } = this.props
    const notScheduled = items.filter(item => item.userScheduled === "")
    return(
      <div className="audit-container">
        <h2>Home Audit</h2>
        <h3>Click an item below to schedule a replacement date</h3>
        <div className="audit-cards-container">
          { notScheduled.map(item => <AuditCard key={ item.id } {...item}/>) }
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  items: state.items
})

export default connect(mapStateToProps)(Audit);

Audit.propTypes = {
  items: PropTypes.array
}