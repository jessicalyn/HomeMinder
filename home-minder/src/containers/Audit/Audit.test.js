import React from 'react';
import { Audit, mapStateToProps } from './Audit';
import { shallow } from 'enzyme';
import { mockHomeData } from '../../mockData'

describe('Audit container', () => {
  describe('Audit', () => {
    let wrapper
    const mockItems = mockHomeData

    beforeEach(() => {
      wrapper = shallow(<Audit items={ mockItems }/>)
    });

    it('should match the snapshot with the data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with an items array', () => {
      const mockState = {
        items: mockHomeData,
        error: "",
        loading: false
      }
      const expected = {
        items: mockHomeData
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  });

})