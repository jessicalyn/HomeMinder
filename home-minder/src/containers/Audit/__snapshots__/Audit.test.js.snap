// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit container Audit should match the snapshot with the data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Audit
    items={
      Array [
        Object {
          "brand": "",
          "category": "Filters",
          "id": "JK1234",
          "lastReplaced": "",
          "model": "",
          "name": "Refrigerator Water Filter",
          "purchaseLocation": "",
          "recommendation": "6 months",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Sponges",
          "id": "IF1234",
          "lastReplaced": "",
          "model": "",
          "name": "Sponge",
          "purchaseLocation": "",
          "recommendation": "2-4 weeks",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Toothbrush",
          "id": "PF123",
          "lastReplaced": "",
          "model": "",
          "name": "Toothbrush",
          "purchaseLocation": "",
          "recommendation": "1-3 months",
          "room": "Bathroom",
          "type": "Replace",
          "userScheduled": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Home Audit
        </h2>,
        <h3>
          Click an item below to schedule a replacement date
        </h3>,
        <div
          className="audit-cards-container"
        >
          <AuditCard
            brand=""
            category="Filters"
            id="JK1234"
            lastReplaced=""
            model=""
            name="Refrigerator Water Filter"
            purchaseLocation=""
            recommendation="6 months"
            room="Kitchen"
            type="Replace"
            userScheduled=""
          />
          <AuditCard
            brand=""
            category="Sponges"
            id="IF1234"
            lastReplaced=""
            model=""
            name="Sponge"
            purchaseLocation=""
            recommendation="2-4 weeks"
            room="Kitchen"
            type="Replace"
            userScheduled=""
          />
          <AuditCard
            brand=""
            category="Toothbrush"
            id="PF123"
            lastReplaced=""
            model=""
            name="Toothbrush"
            purchaseLocation=""
            recommendation="1-3 months"
            room="Bathroom"
            type="Replace"
            userScheduled=""
          />
        </div>,
      ],
      "className": "audit-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home Audit",
        },
        "ref": null,
        "rendered": "Home Audit",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click an item below to schedule a replacement date",
        },
        "ref": null,
        "rendered": "Click an item below to schedule a replacement date",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AuditCard
              brand=""
              category="Filters"
              id="JK1234"
              lastReplaced=""
              model=""
              name="Refrigerator Water Filter"
              purchaseLocation=""
              recommendation="6 months"
              room="Kitchen"
              type="Replace"
              userScheduled=""
            />,
            <AuditCard
              brand=""
              category="Sponges"
              id="IF1234"
              lastReplaced=""
              model=""
              name="Sponge"
              purchaseLocation=""
              recommendation="2-4 weeks"
              room="Kitchen"
              type="Replace"
              userScheduled=""
            />,
            <AuditCard
              brand=""
              category="Toothbrush"
              id="PF123"
              lastReplaced=""
              model=""
              name="Toothbrush"
              purchaseLocation=""
              recommendation="1-3 months"
              room="Bathroom"
              type="Replace"
              userScheduled=""
            />,
          ],
          "className": "audit-cards-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "JK1234",
            "nodeType": "function",
            "props": Object {
              "brand": "",
              "category": "Filters",
              "id": "JK1234",
              "lastReplaced": "",
              "model": "",
              "name": "Refrigerator Water Filter",
              "purchaseLocation": "",
              "recommendation": "6 months",
              "room": "Kitchen",
              "type": "Replace",
              "userScheduled": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "IF1234",
            "nodeType": "function",
            "props": Object {
              "brand": "",
              "category": "Sponges",
              "id": "IF1234",
              "lastReplaced": "",
              "model": "",
              "name": "Sponge",
              "purchaseLocation": "",
              "recommendation": "2-4 weeks",
              "room": "Kitchen",
              "type": "Replace",
              "userScheduled": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "PF123",
            "nodeType": "function",
            "props": Object {
              "brand": "",
              "category": "Toothbrush",
              "id": "PF123",
              "lastReplaced": "",
              "model": "",
              "name": "Toothbrush",
              "purchaseLocation": "",
              "recommendation": "1-3 months",
              "room": "Bathroom",
              "type": "Replace",
              "userScheduled": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Home Audit
          </h2>,
          <h3>
            Click an item below to schedule a replacement date
          </h3>,
          <div
            className="audit-cards-container"
          >
            <AuditCard
              brand=""
              category="Filters"
              id="JK1234"
              lastReplaced=""
              model=""
              name="Refrigerator Water Filter"
              purchaseLocation=""
              recommendation="6 months"
              room="Kitchen"
              type="Replace"
              userScheduled=""
            />
            <AuditCard
              brand=""
              category="Sponges"
              id="IF1234"
              lastReplaced=""
              model=""
              name="Sponge"
              purchaseLocation=""
              recommendation="2-4 weeks"
              room="Kitchen"
              type="Replace"
              userScheduled=""
            />
            <AuditCard
              brand=""
              category="Toothbrush"
              id="PF123"
              lastReplaced=""
              model=""
              name="Toothbrush"
              purchaseLocation=""
              recommendation="1-3 months"
              room="Bathroom"
              type="Replace"
              userScheduled=""
            />
          </div>,
        ],
        "className": "audit-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home Audit",
          },
          "ref": null,
          "rendered": "Home Audit",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click an item below to schedule a replacement date",
          },
          "ref": null,
          "rendered": "Click an item below to schedule a replacement date",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <AuditCard
                brand=""
                category="Filters"
                id="JK1234"
                lastReplaced=""
                model=""
                name="Refrigerator Water Filter"
                purchaseLocation=""
                recommendation="6 months"
                room="Kitchen"
                type="Replace"
                userScheduled=""
              />,
              <AuditCard
                brand=""
                category="Sponges"
                id="IF1234"
                lastReplaced=""
                model=""
                name="Sponge"
                purchaseLocation=""
                recommendation="2-4 weeks"
                room="Kitchen"
                type="Replace"
                userScheduled=""
              />,
              <AuditCard
                brand=""
                category="Toothbrush"
                id="PF123"
                lastReplaced=""
                model=""
                name="Toothbrush"
                purchaseLocation=""
                recommendation="1-3 months"
                room="Bathroom"
                type="Replace"
                userScheduled=""
              />,
            ],
            "className": "audit-cards-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "JK1234",
              "nodeType": "function",
              "props": Object {
                "brand": "",
                "category": "Filters",
                "id": "JK1234",
                "lastReplaced": "",
                "model": "",
                "name": "Refrigerator Water Filter",
                "purchaseLocation": "",
                "recommendation": "6 months",
                "room": "Kitchen",
                "type": "Replace",
                "userScheduled": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "IF1234",
              "nodeType": "function",
              "props": Object {
                "brand": "",
                "category": "Sponges",
                "id": "IF1234",
                "lastReplaced": "",
                "model": "",
                "name": "Sponge",
                "purchaseLocation": "",
                "recommendation": "2-4 weeks",
                "room": "Kitchen",
                "type": "Replace",
                "userScheduled": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "PF123",
              "nodeType": "function",
              "props": Object {
                "brand": "",
                "category": "Toothbrush",
                "id": "PF123",
                "lastReplaced": "",
                "model": "",
                "name": "Toothbrush",
                "purchaseLocation": "",
                "recommendation": "1-3 months",
                "room": "Bathroom",
                "type": "Replace",
                "userScheduled": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
